akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/JoinConfigCompatCheckClusterShardingSpec.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed
+
+import scala.concurrent.duration._
+import scala.collection.{ immutable ⇒ im }
+import akka.actor.ActorSystem
+import akka.cluster.Cluster
+import akka.testkit.{ AkkaSpec, LongRunningTest }
+import com.typesafe.config.{ Config, ConfigFactory }
+
+object JoinConfig {
+
+  val Shards = 2
+
+  val Key = "akka.cluster.sharding.number-of-shards"
+
+  private val baseConfig: Config =
+    ConfigFactory.parseString("""
+      akka.actor.provider = "cluster"
+      akka.cluster.sharding.state-store-mode = "persistence"
+      akka.cluster.configuration-compatibility-check.enforce-on-join = on
+      akka.cluster.jmx.enabled = off
+      akka.coordinated-shutdown.terminate-actor-system = on
+      akka.remote.netty.tcp.port = 0
+      akka.remote.artery.canonical.port = 0
+    """).withFallback(AkkaSpec.testConf)
+
+  def joinConfig(configured: Int): Config =
+    ConfigFactory.parseString(s"$Key = $configured")
+      .withFallback(baseConfig)
+}
+
+abstract class JoinConfigCompatCheckClusterShardingSpec extends AkkaSpec(
+  JoinConfig.joinConfig(JoinConfig.Shards)) {
+
+  protected val duration = 5.seconds
+
+  protected var nodes: Seq[ActorSystem] = Seq.empty
+
+  override protected def beforeTermination(): Unit = nodes.foreach(shutdown(_))
+
+  def join(sys: ActorSystem): Cluster = {
+    nodes :+= sys
+    sys match {
+      case seed if seed eq system ⇒
```

Comment:
What is the best test for this? There is the potential for system.startTime but sometimes in tests that Long can be the same, and there is no unique id.